arkane/encorr/mbac.py

```diff
@@ -0,0 +1,141 @@
+}
+
+
+def get_bac_correction(model_chemistry, coords, nums, multiplicity=1, mol_corr=0.0):
+    """
+    Given the model chemistry, molecular coordinates, atomic numbers,
+    and dictionaries of BAC parameters, return the total BAC correction
+    (should be SUBTRACTED from energy).
+
+    Note that a molecular correction term other than 0 destroys the size
+    consistency of the quantum chemistry method. This correction also
+    requires the multiplicity of the molecule.
+    """
+    alpha = 3.0  # Angstrom^-1
+
+    # Get BAC parameters
+    try:
+        params = data.mbac[model_chemistry]
+    except KeyError:
+        raise Exception('Missing BAC parameters for model chemistry {}'.format(model_chemistry))
+    atom_corr = params['atom_corr']
+    bond_corr_length = params['bond_corr_length']
+    bond_corr_neighbor = params['bond_corr_neighbor']
+
+    # Get single-bonded RMG molecule
+    mol = geo_to_mol(coords, nums)
+
+    # Molecular correction
+    spin = 0.5 * (multiplicity - 1)
+    bac_mol = mol_corr * (spin - sum(atom_spins[atom.element.symbol] for atom in mol.atoms))
+
+    # Atomic correction
+    bac_atom = sum(atom_corr[atom.element.symbol] for atom in mol.atoms)
+
+    # Bond correction
+    bac_bond = 0.0
+    for bond in mol.getAllEdges():
+        atom1 = bond.atom1
+        atom2 = bond.atom2
+        symbol1 = atom1.element.symbol
+        symbol2 = atom2.element.symbol
+
+        # Bond length correction
+        length_corr = (bond_corr_length[symbol1] * bond_corr_length[symbol2]) ** 0.5
+        length = np.linalg.norm(atom1.coords - atom2.coords)
+        bac_bond += length_corr * np.exp(-alpha * length)
+
+        # Neighbor correction
+        for other_atom, other_bond in mol.getBonds(atom1).iteritems():  # Atoms adjacent to atom1
+            if other_bond is not bond:
+                other_symbol = other_atom.element.symbol
+                bac_bond += bond_corr_neighbor[symbol1] + bond_corr_neighbor[other_symbol]
+        for other_atom, other_bond in mol.getBonds(atom2).iteritems():  # Atoms adjacent to atom2
+            if other_bond is not bond:
+                other_symbol = other_atom.element.symbol
+                bac_bond += bond_corr_neighbor[symbol2] + bond_corr_neighbor[other_symbol]
+
+    return (bac_mol + bac_atom + bac_bond) * 4184.0
+
+
+def geo_to_mol(coords, nums):
+    """
+    Convert molecular geometry specified by atomic coordinates and
+    atomic numbers to RMG molecule.
+
+    Use Open Babel for most cases because it's better at recognizing
+    long bonds. Use RMG for hydrogen because Open Babel can't do it for
+    mysterious reasons.
+    """
+    if len(nums) == 2 and all(n == 1 for n in nums):
+        mol = Molecule()
+        mol.fromXYZ(nums, coords)
+    else:
+        xyz = '{}\n\n'.format(len(nums))
+        xyz += '\n'.join('{0}  {1[0]: .10f}  {1[1]: .10f}  {1[2]: .10f}'.format(s, c) for s, c in zip(coords, nums))
```

Comment:
Can you take another look here? Looks like `coords` and `nums` should be switched in the `zip`

Reply:
Oops, great catch.